src/components/big-interactive-pages/editor.tsx

```diff
@@ -522,236 +522,240 @@ export default function Editor({ persistenceState, cookies, roomState }: EditorP
 			}
 		});
 	}, [initialCode]);
-	if(isNewSaveStrat.value && persistenceState.value.kind === PersistenceStateKind.COLLAB && typeof persistenceState.value.game === 'string')
-		return (
-			<RoomPasswordPopup persistenceState={persistenceState} />
-		)
-	else
-		return (
-			<div class={styles.page}>
-				<Navbar persistenceState={persistenceState} roomState={roomState}/>
-
-				<div class={styles.pageMain}>
-					<div className={styles.codeContainer}>
-						<CodeMirror
-							persistenceState={persistenceState}
-							roomState={roomState}
-							class={styles.code}
-							initialCode={initialCode}
-							onEditorView={(editor) => {
-								codeMirror.value = editor;
-								setTimeout(() => foldAllTemplateLiterals(), 100); // Fold after the document is parsed (gross)
-							}}
-							onRunShortcut={onRun}
-							onCodeChange={() => {
+
+	return (
+		isNewSaveStrat.value && persistenceState.value.kind === PersistenceStateKind.COLLAB && typeof persistenceState.value.game === 'string' 
+		? 
+			(<>
+				<RoomPasswordPopup persistenceState={persistenceState} />
+			</>) 
+		:
+			(
+		<div class={styles.page}>
+			<Navbar persistenceState={persistenceState} roomState={roomState}/>
+
+			<div class={styles.pageMain}>
+				<div className={styles.codeContainer}>
+					<CodeMirror
+						persistenceState={persistenceState}
+						roomState={roomState}
+						class={styles.code}
+						initialCode={initialCode}
+						onEditorView={(editor) => {
+							codeMirror.value = editor;
+							setTimeout(() => foldAllTemplateLiterals(), 100); // Fold after the document is parsed (gross)
```

Comment:
What necessitates folding the template literals to begin with?  I don't argue against needing to, just looking to understand (it's probably best to state this explicitly in the comment, tbh)

Reply:
that was done by someone else b4 my changes. it's probabyl because every template literal starts up as unfolded by default

Reply:
okay - just make sure any refactorings you might do don't break any of this stuff :)

Reply:
yup, most of the highlighted lines were just tabs